canonicals.js

```diff
@@ -0,0 +1,53 @@
+var APP = APP || {};
+(function(app) {
+
+    var canonicals = {
+
+        closures: [750486],
+        asynchronous: [14220321],
+        inheritance: [11072556, 7486825]
+
+    };
+
+    var ids = Object.keys(canonicals).map(function(category) {
+        return canonicals[category].join(';');
+    });
+
+    app.xhr.fetch(ids.join(';')).then(function success(questions) {
+        questions.items.forEach(function(question) {
+            Object.keys(canonicals).forEach(function(category) {
+                canonicals[category].forEach(function(q, i) {
+                    //they see me nesting...
+                    if (q === question.question_id) {
+                        canonicals[category][i] = question;
```

Comment:
I'm going to need a crash course on using reduce, will update pull request shortly but I somehow feel I'm not Doing It Right. I ended up doing the same amount of nested loops while increasing the number of LOC. I also didn't quite get how reduce would (pun!) reduce the amount of mapping for building the array of ids sent to the SO API so if you could elaborate that would be very nice.